shared/constants/transaction.ts

```diff
@@ -196,4 +196,18 @@ export enum TokenStandard {
   none = 'NONE',
 }
 
-export const EIP712_PRIMARY_TYPE_PERMIT = 'Permit';
+/**
+ * EIP-712 Permit PrimaryTypes
+ */
+export enum PERMIT_PRIMARY_TYPE {
```

Comment:
Could we go even more general and call this `PrimaryType` assuming we deal with more than just Permit in future?

Should this go in a `shared/constants/signatures.ts`?

As this is an enum, should we go standard `CamelCase`?

Reply:
ah right about the enum naming. A new `shared/constants/signatures.ts` file sounds good to me. Created and moved these in. That said, I couldn't figure out how to get rid of these enum lint errors, so I converted it to a constant until we figure this out 

![CleanShot 2024-07-25 at 18 58 35@2x](https://github.com/user-attachments/assets/dac6503f-0326-4163-817a-dba8d07c843b)

Reply:
Thanks! Updated [8dc3f9f](https://github.com/MetaMask/metamask-extension/pull/26107/commits/8dc3f9f259549819fd9bae64cd71a3e0d77d1a72)

p.s. For some reason, locally, my machine isn't registering the newly file I created for `shared/constants/signatures.ts` as TS. Still seems to work despite errors.